### DISCUSSION TITLE
How to add a Command and CommandParameter to each MenuFlyout item?

### DISCUSSION THREAD

**Discussion by jonjonsson** [AvaloniaUI/Avalonia#14148] [General]
I have the following code and want to add a Command and CommandParameter to each MenuFlyout.Item.  Struggling with this for a while now!

`<Button Content="Add an event" x:Name="TestButton">

	<Button.Flyout>
		<MenuFlyout ItemsSource="{Binding EventTypes}">
			<MenuFlyout.ItemTemplate>
				<DataTemplate DataType="model:TriggerEvents.EventEnumName" >
					<TextBlock Text="{Binding Name}"  />
				</DataTemplate>
			</MenuFlyout.ItemTemplate>
		</MenuFlyout>
	</Button.Flyout>
</Button>`

**Comment by workgroupengineering** [comment]
```xaml
<Button.Flyout>
  <MenuFlyout ItemsSource="{Binding EventTypes}">
    <MenuFlyout.ItemContainerStyle> 
      <ContextMenu.ItemContainerStyle>
            <Style TargetType="MenuItem">
                  <Setter Property="Header"  Value="{Binding Name}" x:DataType="model:TriggerEvents.EventEnumName" />
                  <Setter Property="Command" Value="{Binding $parent.((vm:ParentViewModel)DataContext).YourCommand}" />
                  <Setter Property="CommandParameter"  Value="{Binding Desc}" />
             </Style>
      </ContextMenu.ItemContainerStyle>
    </MenuFlyout.ItemContainerStyle>
  </MenuFlyout>
</Button.Flyout>
```

**Comment by jonjonsson** [comment]
> <ContextMenu.ItemContainerStyle>

Had trouble with the above:

> XamlX.XamlParseException: Dots aren't allowed in type names	

I finally got a working solution with this (excuse the formatting mess, not playing nice):


`<Button Content="Add an Event" x:Name="AllEvents">
    <Button.Flyout>
       <MenuFlyout ItemsSource="{Binding EventTypes}">
            <MenuFlyout.ItemTemplate>
                <DataTemplate DataType="model:TriggerEvents.EventEnumName">

            <MenuItem Header="{Binding Name}" Command="{Binding #AllEvents.((vm:EventsViewModel)DataContext).SelectAddEventCommand}" CommandParameter="{Binding Event}" />

				</DataTemplate>

			</MenuFlyout.ItemTemplate>

		</MenuFlyout>

	</Button.Flyout>

</Button>`
